### DISCUSSION TITLE
Namespace/Scope when importing built-in API modules

### DISCUSSION THREAD

**Discussion by delchev** [eclipse/dirigible#3390] [General]
With the introduction of the ES6 support, it was natural to introduce also the `"namespace"` of the API set provided by the platform. The initial variant is `@dirigible`, which follows the `scoped packages` notation [https://docs.npmjs.com/cli/v10/using-npm/scope](https://docs.npmjs.com/cli/v10/using-npm/scope).

Example:

```javascript
import { response } from "@dirigible/http";

response.println("Hello World!");
```

While in some extent the API provided by Eclipse Dirigible could be considered as `"owned"` by the platform, the context here is a bit different. The `platform` API is actually the only one which can be used in this way, there cannot be others like `@frameworkXXX/...`. This makes it more like THE API of the platform, not one of many, so it is de-facto the SDK for the platform.

Having the above in mind, the proposal is to change the `@dirigible` to `sdk` (without `@`), which will fix:

* implicit suggestion that there could be many libraries imported in this way, which is not true
* `dirigible` is a scope for the `"owned"` by Eclipse Dirigible _standalone_ library, which is not correct
* still preventing collisions with users' projects named like the API modules e.g. `test`, `extensions`, etc.

Example:
  
```javascript
import { response } from "sdk/http";

response.println("Hello World!");
```

Importing libraries from the users' projects remains the same starting with `/`.

Example:
  
```javascript
import { myclass } from "/myproject/mylibrary";

myclass.myfunction();
```

or with `mjs` file extension included:

```javascript
import { myclass } from "/myproject/mylibrary.mjs";

myclass.myfunction();
```

and the combined usage will look like:

```javascript
import { response } from "sdk/http";
import { myclass } from "/myproject/mylibrary";

response.println(myclass.myfunction());
```

**Comment by ThuF** [comment]
Migration from `@dirigible/xxx` to `sdk/xxx` modules to be introduced with:
- https://github.com/eclipse/dirigible/pull/3394
